sdk/identity/azure-identity/azure/identity/_credentials/vscode.py

```diff
@@ -2,149 +2,175 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-import abc
 import os
-import sys
-from typing import cast, Any, Dict, Optional
-import warnings
+import json
+import logging
+from typing import Any, Optional
 
+import msal
 from azure.core.credentials import AccessToken, TokenRequestOptions, AccessTokenInfo
 from azure.core.exceptions import ClientAuthenticationError
+
+from .._auth_record import AuthenticationRecord
 from .._exceptions import CredentialUnavailableError
-from .._constants import AzureAuthorityHosts, AZURE_VSCODE_CLIENT_ID, EnvironmentVariables
-from .._internal import normalize_authority, validate_tenant_id, within_dac
-from .._internal.aad_client import AadClient, AadClientBase
-from .._internal.get_token_mixin import GetTokenMixin
+from .._constants import AZURE_VSCODE_CLIENT_ID
+from .._internal import within_dac
+
 from .._internal.decorators import log_get_token
+from .._internal.utils import get_broker_credential, validate_tenant_id
 
-if sys.platform.startswith("win"):
-    from .._internal.win_vscode_adapter import get_refresh_token, get_user_settings
-elif sys.platform.startswith("darwin"):
-    from .._internal.macos_vscode_adapter import get_refresh_token, get_user_settings
-else:
-    from .._internal.linux_vscode_adapter import get_refresh_token, get_user_settings
 
+VSCODE_AUTH_RECORD_PATH = "~/.azure/ms-azuretools.vscode-azureresourcegroups/authRecord.json"
```

Comment:
Do we also need to search `.Azure` folder?

Reply:
Forgot about that. Added a check for that.